packages/core/src/types.ts

```diff
@@ -589,6 +593,8 @@ export interface InvokeConfig<
   onError?:
     | string
     | SingleOrArray<TransitionConfig<TContext, DoneInvokeEvent<any>>>;
+
+  onEmit?: string | SingleOrArray<TransitionConfig<TContext, EmitEvent<any>>>;
```

Comment:
I have some concerns related to how we are going to type this - but we've seen worse 😉 so hopefully we'll be able to wrangle this.